src/parser.cpp

```diff
@@ -1181,8 +1142,10 @@ namespace ada::parser {
               // If the code point substring from pointer to the end of input does not start with
               // a Windows drive letter and base’s path[0] is a normalized Windows drive letter,
               // then append base’s path[0] to url’s path.
-              if (std::distance(pointer, pointer_end) > 0 && !base_url->path.empty()) {
-                if (!checkers::is_windows_drive_letter({pointer, size_t(pointer_end - pointer)})) {
+              if (std::distance(pointer, pointer_end) > 1 && !base_url->path.empty()) {
+                // is_windows_drive_letter expects a size 2 string prefix.
+                if (!checkers::is_windows_drive_letter({pointer, 2})) {
```

Comment:
`is_windows_drive_letter` also checks for input size, and line 1145 also checks. If we expect it to have at least size 2, we can remove the check inside `is_windows_drive_letter`